### ISSUE TITLE
Online examples not working with HTTPS

### ISSUE THREAD

**Issue by chrismayer** [geoext/geoext2#359]
The online examples under https://geoext.github.io/geoext2/ do not work because ExtJS and OpenLayers are included with HTTP which is blocked by the browsers due to 'Mixed Content'.

**Comment by weskamm** [comment]
thats a critcal one, broken examples need to be fixed quickly, no question.

But im wondering myself, how did you achieve to open the gh-pages with https?
The links from the projects readme are just http, and following them works fine.

Maybe your browser redirected you to the https one? If so, we will have to look into that..

**Comment by chrismayer** [comment]
Even worse: a result of a search engine pointed to the HTTPS version of the gh-pages site.

You are right, if you're following the links in the README everything is fine because it HTTP. But we cant't rely on this IMHO especially when search engines offer the HTTPS version.

Do you have time to care about this @weskamm ?

**Comment by weskamm** [comment]
yes i could spent some time on this.
the real solution would be to use https for every resource, but some of them are not in our hands / control. 
I just had a quick look, the sencha cdn is also available on https, openlayers is hosted on our gh-pages so no problem there, ows.terrestris would need a fixup and there may be other resources i'm missing.

If you find more external resources causing the issue, please report them here

**Comment by weskamm** [comment]
work is ongoing, but cannot continue till friday...

**Comment by marcjansen** [comment]
:thumbsup:

**Comment by weskamm** [comment]
fixed all the resources for the gh-pages, reviewed by @marcjansen , already online.
Fixing the examples in the lib needs a little more effort, as they differ to much in order to apply a patch

**Comment by chrismayer** [comment]
This is again an issue since the online examples have been updated by the GeoExt 2.1.0 examples.

I had a quick look at this. It seems that the ExtJS-resources loaded by this [script](https://github.com/geoext/geoext2/blob/master/examples/include-ext.js), which needs to be adapted. But it seems that the commits 7f16a9e64038d, 057b1953071dcf4, ff16e3f52c2, aa2f354ef26 and ead6943be6d are overwritten by the 2.1.0 release since they have only been pushed to the gh-pages branch.

**Comment by marcjansen** [comment]
It was very bad of us to only commit to the gh-pages branch! So sorry for the confusion this has caused, @chrismayer.

**Comment by chrismayer** [comment]
I tried to catch all HTTP calls which could be replaced by HTTPS. If you still discover any problems please report here.

We should definitely port this patch to the master branch.

**Comment by marcjansen** [comment]
:+1:

**Comment by weskamm** [comment]
Thanks @chrismayer! i checked the examples and found 2 remaining problems:

https://geoext.github.io/geoext2/examples/geocoder/geocoder.html 
uses nominatim over http. A quick test shows that it works with https, so adjusting the url should be enough.

https://geoext.github.io/geoext2/examples/popup/popup.html
has problems with the bbox, as the map shows pink tiles on the first zoomlevel. I know this is not related to this discussion, but i think adjusting / lowering the maxextent should fix this. I currently have no time to fix the service itself, which should be the way to go...

**Comment by chrismayer** [comment]
Thanks for cross-checking @weskamm!
> https://geoext.github.io/geoext2/examples/geocoder/geocoder.html
uses nominatim over http. A quick test shows that it works with https, so adjusting the url should be enough.

Fixed in ea0347f

> https://geoext.github.io/geoext2/examples/popup/popup.html
has problems with the bbox, as the map shows pink tiles on the first zoomlevel. I know this is not related to this discussion, but i think adjusting / lowering the maxextent should fix this. I currently have no time to fix the service itself, which should be the way to go...

As suggested I fixed this by showing an appropriate BBOX (see #371 for the solution).

I opened up an issue for the backport of the fixes regarding this issue to the master branch (#372).

So this can be closed now.
